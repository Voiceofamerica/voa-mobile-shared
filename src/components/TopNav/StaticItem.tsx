
import * as React from 'react'

import { staticItem } from './TopNav.scss'

export interface Props extends React.Props<any> {
  style?: React.CSSProperties
  onClick?: () => void
}

function StaticItem (props: Props) {
  const { children, onClick = () => null, style } = props

  return (
    <button className={staticItem} onClick={() => onClick()} style={style}>
      {children}
    </button>
  )
}

export default StaticItem
