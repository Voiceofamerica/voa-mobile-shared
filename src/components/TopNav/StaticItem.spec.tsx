
import * as React from 'react'
import { create } from 'react-test-renderer'

import StaticItem from './StaticItem'

describe(`<${StaticItem.name} />`, () => {
  describe('shapshots', () => {
    it('should render empty', () => {
      const element = create((
        <StaticItem />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given children', () => {
      const element = create((
        <StaticItem>
          <div>child1</div>
          <div>child2</div>
        </StaticItem>
      ))

      expect(element).toMatchSnapshot()
    })
  })
})
