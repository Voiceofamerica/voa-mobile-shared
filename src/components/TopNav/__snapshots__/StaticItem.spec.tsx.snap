// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticItem /> shapshots should render empty 1`] = `
<button
  className="staticItem"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`<StaticItem /> shapshots should render with the given children 1`] = `
<button
  className="staticItem"
  onClick={[Function]}
  style={undefined}
>
  <div>
    child1
  </div>
  <div>
    child2
  </div>
</button>
`;
